src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -130,6 +140,34 @@ def _create_jobs(self, job_ids):
                 # be set to True.
                 self._running_jobs[job_id] = {"refresh": 0, "job": job}
 
+    def _check_job_list(self, job_id_list: List[str] = []):
```

Comment:
New function for job ID cleaning and deduping.

Reply:
How often do we see duplication? Or is this just being cautious?
I guess I could see the lookup for an individual job, plus that job again as the child of some other job.

Reply:
mostly just being cautious -- the main example I thought of was the child job case you mention.